src/runtime_src/core/pcie/emulation/cpu_em/generic_pcie_hal2/shim.h

```diff
@@ -246,7 +247,7 @@ namespace xclcpuemhal2 {
       uint64_t mReqCounter;
       FeatureRomHeader mFeatureRom;
       boost::property_tree::ptree mPlatformData;
-      std::map<std::string, std::string> mQueryTable;
+      std::map<key_type, std::string> mQueryTable;
```

Comment:
At the end I wonder whether an array or a `std::vector<std::string>` indexed by an equivalent of an `enum` is not what we want here...

Reply:
std::vector<std::string> not required in this case as we dont need to maintain the vector with max size of enum.